rspace/src/main/scala/coop/rchain/rspace/ReplayRSpace.scala

```diff
@@ -204,13 +213,25 @@ class ReplayRSpace[F[_], C, P, E, A, R, K](store: IStore[F, C, P, A, K], branch:
       implicit m: Match[P, E, A, R]
   ): F[Either[E, Option[(ContResult[C, P, K], Seq[Result[R]])]]] =
     contextShift.evalOn(scheduler) {
-      produceLockF(channel) {
-        lockedProduce(channel, data, persist, sequenceNumber)
-      }
+      for {
+        span <- metricsF.span(produceSpanLabel)
+        _    = span.mark("before-produce-lock")
+        result <- produceLockF(channel) {
+                   lockedProduce(channel, data, persist, sequenceNumber, span)
+                 }
+        _ = span.mark("post-produce-lock")
+        _ = span.close()
```

Comment:
what happens if there's an error and we don't close the span? The Bracket/Resource thing for spans might actually be necessary depending on the answer :)

Reply:
span does not send